src/Interpreters/ClusterProxy/SelectStreamFactory.cpp

```diff
@@ -277,29 +267,22 @@ void SelectStreamFactory::createForShard(
             }
 
             if (try_results.empty() || local_delay < max_remote_delay)
```

Comment:
@alexey-milovidov Do you happen to know, why is it decided by the most stale replica delay even when we have up-to-date replicas?